enterprise/frontend/src/embedding-sdk/test/sdk-compatibility-warning.unit.spec.tsx

```diff
@@ -0,0 +1,116 @@
+import { render } from "@testing-library/react";
+import fetchMock from "fetch-mock";
+
+import {
+  setupCurrentUserEndpoint,
+  setupPropertiesEndpoints,
+} from "__support__/server-mocks";
+import { waitForLoaderToBeRemoved } from "__support__/ui";
+import {
+  MetabaseProvider,
+  defineEmbeddingSdkConfig,
+} from "embedding-sdk/components/public";
+import {
+  createMockSettings,
+  createMockTokenFeatures,
+  createMockUser,
+  createMockVersion,
+} from "metabase-types/api/mocks";
+
+import { getEmbeddingSdkVersion } from "../config";
+
+// TODO: extract this common setup to a shared util
+const METABASE_INSTANCE_URL = "path:";
+const AUTH_PROVIDER_URL = "http://auth-provider/metabase-sso";
+
+const defaultAuthUriConfig = defineEmbeddingSdkConfig({
+  metabaseInstanceUrl: METABASE_INSTANCE_URL,
+  authProviderUri: AUTH_PROVIDER_URL,
+  fetchRequestToken: _ =>
+    Promise.resolve({
+      id: "123",
+      exp: Number.MAX_SAFE_INTEGER,
+    }),
+});
+
+jest.mock("../config", () => ({
+  getEmbeddingSdkVersion: jest.fn(),
+}));
+
+const setup = async ({
+  sdkVersion,
+  mbVersion,
+}: {
+  sdkVersion: string;
+  mbVersion: string;
+}) => {
+  (getEmbeddingSdkVersion as jest.Mock).mockReturnValue(sdkVersion);
+  setupPropertiesEndpoints(
+    createMockSettings({
+      "token-features": createMockTokenFeatures({
+        embedding_sdk: true,
+      }),
+      version: createMockVersion({ tag: mbVersion }),
+    }),
+  );
+  setupCurrentUserEndpoint(createMockUser({ id: 1 }));
+
+  render(
+    <MetabaseProvider config={defaultAuthUriConfig}>
+      <div>Hello</div>
+    </MetabaseProvider>,
+  );
+  await waitForLoaderToBeRemoved();
+};
+
+let consoleWarnSpy: jest.SpyInstance;
+
+const getWarnMessages = (): string[] =>
+  consoleWarnSpy.mock.calls.map(call => call.join(" "));
+
+describe("SDK auth errors", () => {
+  beforeEach(() => {
+    fetchMock.reset();
+
+    consoleWarnSpy = jest.spyOn(console, "warn").mockImplementation(() => {});
+    (getEmbeddingSdkVersion as jest.Mock).mockClear();
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe("SDK version compatibility", () => {
+    it("should show a message when the SDK version is not compatible with the Metabase version", async () => {
```

Comment:
These tests are for positive case, negative case, and for when the version is unknown (not sure if it's actually possible, but the code defaults the version to that)